artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPLargeMessage.java

```diff
@@ -622,7 +624,15 @@ public long getPersistentSize() throws ActiveMQException {
 
    @Override
    public void reencode() {
+      reEncoded = true;
```

Comment:
Why isnt the reencoding happening immediately like for other messages and expected behaviour, e.g. other areas of code and plugins are expecting the message to rencoded by calling this without doing extra work.

Reply:
It would be fairly difficult to re-encode the message on the large message file.
For AMQP Large messages, I'm instead just flagging the message as re-encoded, and I would replace the stream during the delivery.